packages/kbn-es-archiver/src/lib/docs/index_doc_records_stream.ts

```diff
@@ -68,7 +69,7 @@ export function createIndexDocRecordsStream(
   }
 
   return new Writable({
-    highWaterMark: 300,
+    highWaterMark: parseInt((process.env.HIGH_WATER_MARK as string) ?? 5000, 10),
```

Comment:
Same here: I know we discussed it internally, but like we didn't know answer to "why was it set to 300", no one outside Appex-QA will understand about HIGH_WATER_MARK variable and new default value.